### PR TITLE
fix(lib): support API directive queries

### LINKED ISSUE #266: Expose `fxFlex` directive class to access via ContentChildren
## Requirement
I need to access `fxFlex` via `ContentChildren` from a adjacent directive to dynamically modify widths.

## Use Case
I want to create a responsive split component that leverages flex-layout for the layout.

use

#### Usage
```javascript
<div fxLayout="row" ngxSplit="row">
  <div fxFlex="30%" ngxSplitArea>
    Left
  </div>
  <div fxFlex="5px">
    <a href="#" ngxSplitHandle>...</a>
  </div>
  <div fxFlex="70%" ngxSplitArea>
    Right
  </div>
</div>
```

#### split.directive.ts
```javascript
import { 
  Directive, Input, ChangeDetectionStrategy, ContentChild, 
  ContentChildren, AfterContentInit, QueryList
} from '@angular/core';
import { SplitAreaDirective } from './split-area.directive';
import { SplitHandleDirective } from './split-handle.directive';

@Directive({
  selector: '[ngxSplit]',
  changeDetection: ChangeDetectionStrategy.OnPush,
  host: {
    class: 'ngx-split'
  }
})
export class SplitDirective implements AfterContentInit {

  /*tslint:disable*/
  @Input('ngxSplit') 
  direction: string = 'vertial';
  /*tslint:enable*/

  @ContentChild(SplitHandleDirective) handle: SplitHandleDirective;
  @ContentChildren(SplitAreaDirective) areas: QueryList<SplitAreaDirective>;

  ngAfterContentInit(): void {
    console.log('ha', this.handle, this.areas); //<-- Need it here
    this.handle.drag.subscribe(pos => this.onDrag(pos));
  }

  onDrag({ x, y }): void {
    this.areas.forEach(area => {
      console.log('area', area); // TODO: Resize here
    });
  }

}
```
#### split-area.directive.ts
```javascript
import { Directive, Input, ChangeDetectionStrategy, ContentChildren, QueryList } from '@angular/core';

@Directive({
  selector: '[ngxSplitArea]',
  changeDetection: ChangeDetectionStrategy.OnPush,
  host: {
    class: 'ngx-split-area'
  }
})
export class SplitAreaDirective {

  @ContentChildren('[fxFlex]') layouts: QueryList<any>;

}
```

#### split-handle.directive.ts
```javascript
import { Directive, ElementRef, Output, ChangeDetectionStrategy } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { Subscription } from 'rxjs/Subscription';
import 'rxjs/add/operator/takeUntil';
import 'rxjs/add/observable/fromEvent';
import 'rxjs/add/operator/takeUntil';
import 'rxjs/add/operator/switchMap';

@Directive({
  selector: '[ngxSplitHandle]',
  changeDetection: ChangeDetectionStrategy.OnPush,
  host: {
    class: 'ngx-split-handle'
  }
})
export class SplitHandleDirective {

  @Output() drag: Observable<{ x: number, y: number }>;

  constructor(ref: ElementRef) {
    const getMouseEventPosition = (event: MouseEvent) => ({ x: event.clientX, y: event.clientY });

    const mousedown$ = Observable.fromEvent(ref.nativeElement, 'mousedown').map(getMouseEventPosition);
    const mousemove$ = Observable.fromEvent(document, 'mousemove').map(getMouseEventPosition);
    const mouseup$ = Observable.fromEvent(document, 'mouseup');
    
    this.drag = mousedown$.switchMap(mousedown =>
        mousemove$.map(mousemove => ({
          x: mousemove.x - mousedown.x,
          y: mousemove.y - mousedown.y
        }))
        .takeUntil(mouseup$)
      );
  }

}
```

### PR DESCRIPTION
**Pull request by ThomasBurleson**
* support DI of flexbox/api Directive classes
* implement Directive method `activatedValue` getter/setter
  * to imperatively read current activated input value
  * to write API values with immediate style updates
* implement demo using splitters (ngxSplit) with fxLayout and fxFlex layouts

> Based on issues and ideas by @amcdnl + [ngx-u](https://github.com/swimlane/ngx-ui)

Fixes #266.

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
screen shot 2017-05-18 at 7 19 36 pm

**Comment by ThomasBurleson** [comment]
@kara - please review.

**Comment by amcdnl** [comment]
@ThomasBurleson - This looks awesome!

**Comment by ThomasBurleson** [comment]
@jelbourn - please presubmit.

**Comment by jelbourn** [comment]
+ @mmalerba should take a look

File: src/lib/flexbox/api/flex.spec.ts [hunk 1]
```diff
@@ -126,15 +128,15 @@ describe('flex directive', () => {
       `);
       fixture.detectChanges();
       let element = queryFor(fixture, "[fxFlex]")[0].nativeElement;
-      let parent  = queryFor(fixture, ".test")[0].nativeElement;
+      let parent = queryFor(fixture, ".test")[0].nativeElement;
```
Comment (mmalerba) [inline src/lib/flexbox/api/flex.spec.ts]: single quotes

File: src/lib/flexbox/api/flex.spec.ts [hunk 2]
```diff
@@ -149,11 +151,11 @@ describe('flex directive', () => {
       `);
       fixture.detectChanges();
       let element = queryFor(fixture, "[fxFlex]")[0].nativeElement;
-      let parent  = queryFor(fixture, ".parent")[0].nativeElement;
+      let parent = queryFor(fixture, ".parent")[0].nativeElement;
```
Comment (mmalerba) [inline src/lib/flexbox/api/flex.spec.ts]: single quotes

File: src/lib/flexbox/api/base.ts [hunk 3]
```diff
@@ -65,10 +91,17 @@ export abstract class BaseFxDirective implements OnDestroy {
   // *********************************************
 
   /**
+   * Abstract method...
```
Comment (ThomasBurleson) [inline src/lib/flexbox/api/base.ts]: Thanks for drawing my attention to this. This was only a partial (in fact poor) solution.
I found another more robust approach.

**Comment by ThomasBurleson** [comment]
@mmalerba - updated with all your suggested changes. `set activatedInput(value)` now triggers `ngOnChanges()`; see https://github.com/angular/flex-layout/pull/290/files#diff-abac303936dc314fed0f18d5429ac42f

**Pull request merged**